src/DssSpell.t.sol

```diff
@@ -490,4 +592,431 @@ contract DssSpellTest is DssSpellTestBase {
+        assertEq(dai.balanceOf(address(rwaurn_014)), drawAmount - pushAmount, "RWA014: Dai not sent to destination after push()");
+
+        // END
+        GodMode.setWard(address(end), address(this), 1);
+        end.cage();
+        end.cage("RWA014-A");
+
+        end.skim("RWA014-A", address(rwaurn_014));
+
+        (ink, art) = vat.urns("RWA014-A", address(rwaurn_014));
+        uint256 skimmedInk = drawAmount / 500_000_000;
+        assertEq(ink, 1 * WAD - skimmedInk, "RWA014: wrong ink in urn after skim");
+        assertEq(art, 0, "RWA014: wrong art in urn after skim");
+        vm.warp(block.timestamp + end.wait());
+
+        // Removing the surplus to allow continuing the execution.
+        vm.store(
+            address(vat),
+            keccak256(abi.encode(address(vow), uint256(5))),
+            bytes32(uint256(0))
+        );
+
+        end.thaw();
+
+        end.flow("RWA014-A");
+
+        GodMode.setBalance(address(dai), address(this), 1_000_000 * WAD);
+        dai.approve(address(daiJoin), 1_000_000 * WAD);
+        daiJoin.join(address(this), 1_000_000 * WAD);
+
+        vat.hope(address(end));
+        end.pack(1_000_000 * WAD);
+
+        // Check DAI redemption after "cage()"
+        assertEq(vat.gem("RWA014-A", address(this)), 0, "RWA014: wrong vat gem");
+        assertEq(rwagem_014.balanceOf(address(this)), 0, "RWA014: wrong gem balance");
+        end.cash("RWA014-A", 1_000_000 * WAD);
+        assertGt(vat.gem("RWA014-A", address(this)), 0, "RWA014: wrong vat gem after cash");
+        assertEq(rwagem_014.balanceOf(address(this)), 0, "RWA014: wrong gem balance after cash");
+        rwajoin_014.exit(address(this), vat.gem("RWA014-A", address(this)));
+        assertEq(vat.gem("RWA014-A", address(this)), 0, "RWA014: wrong vat gem after exit");
+        assertGt(rwagem_014.balanceOf(address(this)), 0, "RWA014: wrong gem balance after exit");
+    }
+
+    function testRWA014_SPELL_LOCK() public {
+        (uint256 pink, uint256 part) = vat.urns("RWA014-A", address(rwaurn_014));
+        uint256 prevBalance = rwagem_014.balanceOf(address(rwaurn_014.gemJoin()));
+
+        assertEq(part, 0, "RWA014/bad-art-before-spell");
+        assertEq(pink, 0, "RWA014/bad-ink-before-spell");
+
+        uint256 lockAmount = 1 * WAD;
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        // Check if spell lock whole unit of RWA014 Token to the Urn
+        assertEq(rwagem_014.balanceOf(address(rwaurn_014.gemJoin())), prevBalance + lockAmount, "RWA014/spell-do-not-lock-rwa014-token");
+        
+        (uint256 ink, uint256 art) = vat.urns("RWA014-A", address(rwaurn_014));
+        assertEq(art, 0, "RWA014/bad-art-after-spell");
+        assertEq(ink, lockAmount, "RWA014/bad-ink-after-spell"); // Whole unit of collateral is locked
+    }
+
+    function testSparkLendCollateralOnboarding() public {
```

Comment:
Hey @hexonaut did you do tests for LTV Liq threshold and reward and liq bonus? 

DC tests look fine but I don't see any on the other items in the sheet

Reply:
Those could be added.

Reply:
done in https://github.com/makerdao/spells-goerli/pull/198